### DISCUSSION TITLE
Dealing with observation-level attributes

### DISCUSSION THREAD

**Discussion by LucyGwilliamAdmin** [open-sdg/open-sdg#1459] [Ideas]
An observation level attribute within SDMX is COMMENT_OBS, which I think is sort of an observation level footnote.

How could we deal with these - currently they are completely hidden but the information they hold is quite important.

For example, [here](https://github.com/sdg-kyrgyzstan/open-sdg-data-starter/blob/develop/data/2.2.1.xml#L105) you can see that the education level is of the mother. While the [indicator page](https://sdg-kyrgyzstan.github.io/open-sdg-site-starter/en/2-2-1/) doesn't mention it at all

Similarly, [here](https://github.com/sdg-kyrgyzstan/open-sdg-data-starter/blob/develop/data/6.2.1.xml#L66) you can see that education level is of the head of household, but not clear from [indicator page](https://sdg-kyrgyzstan.github.io/open-sdg-site-starter/en/6-2-1/).

The footnotes could just be added manually using the metadata field but just thinking whether there is something we can do.

My initial thinking is if there is only one comment_obs for a disaggregation, the dropdown label could be [DISAGGREGATION NAME] ([COMMENT_OBS]).

or if there is two comment_obs for a disaggregation, there could be two separate dropdown menus:

- [DISAGGREGATION NAME] ([COMMENT_OBS 1])

- [DISAGGREGATION NAME] ([COMMENT_OBS 2])

See comments on #1315 for further information and info about user needs

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning did you have any thoughts on this?

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin I haven't had a chance to think about this, but what you describe sounds good. And it does seem like an important feature to add to the SDMX inputs in sdg-build. I believe "CUSTOM_BREAKDOWN" also shows up in the SDMX as an attribute, and this might allow us to use those too.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning if discussing SDMX things with Abdulla, this might be one worth bringing up too

**Comment by brockfanning** [comment]
Will give this some thought and lay out some options. And yes we should bring up with Abdulla.

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin in your description above you mention an observation comment "for a disaggregation". But in my imagination the observation comments are "for a value" but not a particular disaggregation. But maybe, in practice, a particular observation comment tends to only appear when a particular disaggregation is being used? For example:

Year|Urbanisation|Observation comment|Value
---|---|---|---
2019||Taken from 2019 census|40
2019|Rural|Estimated from global data|50
2019|Urban|Estimated from global data|60

So in that case, the dropdown for Urbanisation would show as:

**Urbanisation (Estimated from global data)**
* Rural
* Urban

And then if the disaggregation had more than one observation comment, like this:

Year|Urbanisation|Observation comment|Value
---|---|---|---
2019||Taken from 2019 census|40
2019|Rural|Taken from 2019 census|50
2019|Urban|Estimated from global data|60

Then there would two dropdowns for Urbanisation, showing like:

**Urbanisation (Taken from 2019 census)**
* Rural

**Urbanisation (Estimated from global data)**
* Urban

Do I understand that right?

**Comment by brockfanning** [comment]
For an example of another platform's approach, see how the .Stat suite handles it using [asterisks with tooltips in each cell of the table](http://camstat.nis.gov.kh/vis?locale=en&start=0&constraints[0]=6nQpoAP%2C0%7CSDG%20indicators%23SDG%23&facet=6nQpoAP&dataflow[datasourceId]=CamStat-stable&dataflow[dataflowId]=DF_SDG_KH&dataflow[agencyId]=KH_NIS&dataflow[version]=1.2&period=2008%2C2020&frequency=A&dataquery=A..SI_POV_NAHC............&layout[header]=TIME_PERIOD&layout[rows]=REF_AREA%2CURBANISATION). But that platform does not attempt to display these attributes on the chart displays.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning @phillipgardiner we had this down as 1.3.0 but obviously it didn't go in. I'm thinking it probably needs some user research to decide how the comment_obs would be displayed and also it overlaps with other SDMX fields so doubt could go into 1.5.0, right?

Removing 1.3.0 label though

**Comment by phillipgardiner** [comment]
@LucyGwilliamAdmin Sorry again can't comment as don't haven't totally got my head around this one, but my initial reaction is that yes some user research would be required to either help come up with potential options or evaluate options already suggested.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning I think we need to widen this out so that it documents other SDMX needs too

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin Maybe we could rename it to "Observation-level attributes"?

I think that the most difficult scenario is probably the one we should focus on: when every observation has a different value in an observation-level attribute. For example, if COMMENT_OBS is used, but every observation has a different value. Something like this:

Year | COMMENT_OBS | Value
--- | --- | ---
2015 | Comment 1 | 100
2016 | Comment 2 | 200
2017 | Comment 3 | 300
2018 | Comment 4 | 400
2019 | Comment 5 | 500
2020 | Comment 6 | 600
2021 | Comment 7 | 700

I think what we really want here is a "headline" that displays the COMMENT_OBS (in some way) for each data point. Since it is impossible to display stuff on/in a single point on a chart, one obvious approach might be to put under each observation in the tooltip that pops up. Like right here:

![screenshot-onsdigital github io-2021 07 29-10_46_00](https://user-images.githubusercontent.com/1319083/127513284-970d20a2-d47a-4ca5-a917-8b3c5a9e4f16.png)

That might look a little repetitive, though, in the case that the COMMENT_OBS are all the same, like this:

Year | COMMENT_OBS | Value
--- | --- | ---
2015 | Comment 1 | 100
2016 | Comment 1 | 200
2017 | Comment 1 | 300
2018 | Comment 1 | 400
2019 | Comment 1 | 500
2020 | Comment 1 | 600
2021 | Comment 1 | 700

So if that were a common use-case, then maybe we could add some special code so that it only displays the COMMENT_OBS once, below all the observations, instead of repeating it below each one separately.

Then there is the question of the tables. I have a feeling we would need to implement tooltips for the tables too, like .Stat does in the link above.

**Comment by WohnungNDS** [comment]
Any progess in here?

![image](https://user-images.githubusercontent.com/67795376/139674743-19d06b7a-d483-4360-804c-b1cc9432398d.png)

I Also add an exmple.xlsx....
[Example.xlsx](https://github.com/open-sdg/open-sdg/files/7453344/Example.xlsx)

**Reply by christoph-LSN** [reply]
@LucyGwilliamAdmin @brockfanning @MoBosse 

I tried to use COMMENT_OBS. It works more ore less. In the chart tab the tooltip shows "Statistisch unsicher" correctly, but the bar is only shown if I tick the disagg COMMENT_OBS 

![image](https://user-images.githubusercontent.com/94443931/146572028-8a1b405e-7900-434f-8f73-ef9cea0d53dc.png)

as mentioned by @brockfanning it doesnot work in the table tab. 

![image](https://user-images.githubusercontent.com/94443931/146571825-da19465d-744f-4795-8dad-a05e600718bf.png)

[Exmaple Indicator](https://christoph-lsn.github.io/MT_Site/6-3-1/)
[CVS Data](https://github.com/christoph-LSN/IM_Data/blob/develop/data/indicator_6-3-1.csv)

Is there any progress in this?

**Comment by christoph-LSN** [comment]
@brockfanning @LucyGwilliamAdmin another issue that is important for my migration report..... is there any progress in here?
